Content.Server/Storage/EntitySystems/EntityStorageSystem.cs

```diff
@@ -59,12 +114,239 @@ public void EmptyContents(EntityUid uid, EntityStorageComponent? component = nul
+
+            return false;
+        }
+
+        var ev = new StorageOpenAttemptEvent();
+        RaiseLocalEvent(target, ev, true);
+
+        return !ev.Cancelled;
+    }
+
+    public bool CanClose(EntityUid target, bool silent = false)
+    {
+        var ev = new StorageCloseAttemptEvent();
+        RaiseLocalEvent(target, ev, silent);
+
+        return !ev.Cancelled;
+    }
+
+    public bool AddToContents(EntityUid toAdd, EntityUid container, EntityStorageComponent? component = null)
+    {
+        if (!Resolve(container, ref component))
+            return false;
+
+        if (toAdd == container)
+            return false;
+
+        if (TryComp<IPhysBody>(toAdd, out var phys))
+            if (component.MaxSize < phys.GetWorldAABB().Size.X || component.MaxSize < phys.GetWorldAABB().Size.Y)
+                return false;
+
+        return Insert(toAdd, container, component);
+    }
+
+    public bool CanFit(EntityUid toInsert, EntityUid container)
+    {
+        // conditions are complicated because of pizzabox-related issues, so follow this guide
+        // 0. Accomplish your goals at all costs.
+        // 1. AddToContents can block anything
+        // 2. maximum item count can block anything
+        // 3. ghosts can NEVER be eaten
+        // 4. items can always be eaten unless a previous law prevents it
+        // 5. if this is NOT AN ITEM, then mobs can always be eaten unless unless a previous
+        // law prevents it
+        // 6. if this is an item, then mobs must only be eaten if some other component prevents
+        // pick-up interactions while a mob is inside (e.g. foldable)
+        var attemptEvent = new InsertIntoEntityStorageAttemptEvent();
+        RaiseLocalEvent(toInsert, attemptEvent);
+        if (attemptEvent.Cancelled)
+            return false;
+
+        // checks
+        // TODO: Make the others sub to it.
+        var targetIsItem = HasComp<SharedItemComponent>(toInsert);
+        var targetIsMob = HasComp<SharedBodyComponent>(toInsert);
+        var storageIsItem = HasComp<SharedItemComponent>(container);
+
+        var allowedToEat = targetIsItem;
+
+        // BEFORE REPLACING THIS WITH, I.E. A PROPERTY:
+        // Make absolutely 100% sure you have worked out how to stop people ending up in backpacks.
+        // Seriously, it is insanely hacky and weird to get someone out of a backpack once they end up in there.
+        // And to be clear, they should NOT be in there.
+        // For the record, what you need to do is empty the backpack onto a PlacableSurface (table, rack)
+        if (targetIsMob)
+        {
+            if (!storageIsItem)
+                allowedToEat = true;
+            else
+            {
+                var storeEv = new StoreThisAttemptEvent();
+                RaiseLocalEvent(container, storeEv);
+                allowedToEat = !storeEv.Cancelled;
```

Comment:
this event needs to be opt-in, rather than opt-out. This is currently responsible for pizza boxes eating people. IMO, just give this a a `handled` bool, so that allowedToEat = `ev.Handled && ! ev.Cancelled`, then have foldable & whatever else uses this mark the event as handled.

Also the even name should be changed to be more descriptive, its not a generic storage check, but specifically checking if a mob should be allowed to be inserted into an item-container. Probably something like `MobContainerStoreAttemptEvent` if fine, though I'm partial to `YouArePossibleAboutToStoreAPlayerInAContainerAreYouReallySureAboutThisAttemptEvent`